server/routes/api/campaigns.js

```diff
@@ -0,0 +1,242 @@
+// @desc Get all campaigns of logged in user. Requires no params
+// @access Authenticated Users
+router.get(
+  "/",
+  passport.authenticate("jwt", { session: false }),
+  async (req, res) => {
+    try {
+      const userId = req.user.id;
+
+      const campaigns = await Campaign.find({ ownedBy: userId });
+      res.json(campaigns);
+    } catch (error) {
+      console.log(error);
+      res.json({ error: "Error getting campaigns" });
+    }
+  }
+);
+
+// @route GET /api/campaigns/campaign
+// @desc Get specified campaign of logged in user. Requires 'campaignId'
+// @access Authenticated Users
+router.get(
+  "/campaign/:campaignId",
+  passport.authenticate("jwt", { session: false }),
+  async (req, res) => {
+    try {
+      const userId = req.user.id;
+      const campaignId = req.params.campaignId;
+
+      const campaign = await Campaign.find({
+        _id: campaignId,
+        ownedBy: userId
+      });
+
+      if (!campaign || campaign.length === 0) {
+        res
+          .status(404)
+          .send({ id: `Campaign with id ${campaignId} is not found` });
+      } else {
+        res.json(campaign);
+      }
+    } catch (error) {
+      console.log(error);
+      res.json({ error: "Error getting campaign" });
+    }
+  }
+);
+
+// @route DELETE /api/campaigns/campaign
+// @desc Remove campaign. Requires campaignId
+// @access Authenticated Users
+router.delete(
+  "/campaign/:campaignId",
+  passport.authenticate("jwt", { session: false }),
+  async (req, res) => {
+    try {
+      const userId = req.user.id;
+      console.log();
+      const campaignId = req.params.campaignId;
+
+      const results = await Campaign.deleteOne({
+        _id: campaignId,
+        ownedBy: userId
+      });
+
+      res.json({ id: `${results} removed` });
+    } catch (error) {
+      console.log(error);
+      res.json({ error: "Error deleting campaign" });
+    }
+  }
+);
+
+// @route POST /api/campaigns/prospects
+// @desc Add prospects to a Campaign. Requires 'campaignId' and 'prospects' (array of Prospect ids)
+// @access Authenticated Users
+router.post(
+  "/prospects",
+  passport.authenticate("jwt", { session: false }),
+  async (req, res) => {
+    try {
+      const userId = req.user.id;
+      const { campaignId, prospectIds } = req.body;
+
+      // Check if any prospects being added are already in the campaign
+      const uniqueProspectIds = validateAddProspectsInput(
+        campaignId,
+        prospectIds
+      );
+
+      const campaign = await Campaign.findOne({
+        _id: campaignId,
+        ownedBy: userId
+      });
+
+      if (!campaign || campaign.length === 0) {
+        res
+          .status(404)
+          .send({ id: `Campaign with id ${campaignId} is not found` });
+      } else {
+        // turn array of ids into array of objects for adding to campaign.prospects
+        const arrayOfProspectObj = uniqueProspectIds.map(prospectId => {
+          return {
+            prospectId: prospectId
```

Comment:
I think you changed it to _id and here you used prospectId

Reply:
Good spot. _id now changed back to prospectId